src/main/java/teammates/ui/webapi/action/BinCourseAction.java

```diff
@@ -33,7 +33,7 @@ public ActionResult execute() {
         String idOfCourseToBin = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
         try {
             CourseAttributes courseAttributes = logic.getCourse(idOfCourseToBin);
-            logic.moveCourseToRecycleBin(idOfCourseToBin);
+            courseAttributes.deletedAt = logic.moveCourseToRecycleBin(idOfCourseToBin);
```

Comment:
My apologies. I thought we will return the updated course through the API.

Looks like your change is necessary. In addition, don't forget to add Java test for it.